sound/soc/sof/control.c

```diff
@@ -439,24 +440,38 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 		return -EINVAL;
 	}
 
-	if (copy_from_user(cdata->data, tlvd->tlv, header.length))
+	/* Allocate temporary buffer for new cdata */
+	new_cdata = kzalloc(sizeof(*new_cdata) + header.length, GFP_KERNEL);
+	if (!new_cdata)
+		return -ENOMEM;
+
+	if (copy_from_user(new_cdata->data, tlvd->tlv, header.length)) {
+		kfree(new_cdata);
 		return -EFAULT;
+	}
 
-	if (cdata->data->magic != SOF_ABI_MAGIC) {
+	if (new_cdata->data->magic != SOF_ABI_MAGIC) {
 		dev_err_ratelimited(sdev->dev,
 				    "error: Wrong ABI magic 0x%08x.\n",
-				    cdata->data->magic);
+				    new_cdata->data->magic);
+		kfree(new_cdata);
 		return -EINVAL;
 	}
 
-	if (SOF_ABI_VERSION_INCOMPATIBLE(SOF_ABI_VERSION, cdata->data->abi)) {
-		dev_err_ratelimited(sdev->dev, "error: Incompatible ABI version 0x%08x.\n",
-				    cdata->data->abi);
+	if (SOF_ABI_VERSION_INCOMPATIBLE(SOF_ABI_VERSION,
+					 new_cdata->data->abi)) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: Incompatible ABI version 0x%08x.\n",
+				    new_cdata->data->abi);
+		kfree(new_cdata);
```

Comment:
Was about to comment that why modify 80+ lines but oh wait, you are modifying s/cdata/new_cdata/ ... -> Ok